### PR TITLE
36 register route

### LINKED ISSUE #36: create POST /api/v1/auth/register route
[Auth](https://github.com/harakeke-2019/jwt-auth)

Create a server/db/users.js that has a createUser db function.
Follow the rest of the instructions linked above.

Update the [wiki](https://github.com/harakeke-2019/EDAMAME/wiki/Server-Routes) accordingly.

### PR DESCRIPTION
**Pull request by benlittlenz**
Changed method for authentication.
Now using Don's suggested way with libsodium , jsonwebtoken and dotenv.

Register route now returns a unique token for user if user.name and user.surename does not exist in the database.

closes #36

### CODE REVIEW

**Comment by benlittlenz** [comment]
The wiki needs to be updated.

**Comment by emilyparkes** [comment]
@nzbenji can you please update the wiki then :)

File: server/server.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
 const path = require('path')
 const express = require('express')
-
 const server = express()
 
 const assessmentsRoutes = require('./routes/assessments')
 const cohortsRoutes = require('./routes/cohorts')
 const usersRoutes = require('./routes/users')
+const auth = require('./routes/auth')
```
Comment (emilyparkes) [inline server/server.js]: Please follow code convention of `const authRoutes = require('./routes/auth')`

File: server/server.js [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 
 server.use(express.json())
 server.use(express.static(path.join(__dirname, './public')))
 
 server.use('/api/v1/assessments', assessmentsRoutes)
 server.use('/api/v1/cohorts', cohortsRoutes)
 server.use('/api/v1/users', usersRoutes)
+server.use('/api/v1/auth', auth)
```
Comment (emilyparkes) [inline server/server.js]: Please follow code convention of `server.use('/api/v1/auth', authRoutes)`

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: server/db/users.js [hunk 3]
```diff
@@ -1,29 +1,34 @@
-const connection = require('./')
+const environment = process.env.NODE_ENV || 'development'
```
Comment (emilyparkes) [inline server/db/users.js]: Please use what you removed in line 1 as the next few lines are from the index file it imports.

**Review by emilyparkes — CHANGES_REQUESTED** [review]

**Comment by benlittlenz** [comment]
Made amendments to files

**Review by emilyparkes — DISMISSED** [review]

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**